packages/cloudflare/src/utils/generate-routes-json.ts

```diff
@@ -0,0 +1,267 @@
+		}
+	}
+
+	getAllPaths(): string[][] {
+		const allPaths: string[][] = [];
+		this.dfs(this.root, [], allPaths);
+		return allPaths;
+	}
+}
+
+export default async function (
+	_config: AstroConfig,
+	routes: RouteData[],
+	pages: {
+		pathname: string;
+	}[],
+	redirects: RoutePart[][][],
+	includeExtends:
+		| {
+				pattern: string;
+		  }[]
+		| undefined,
+	excludeExtends:
+		| {
+				pattern: string;
+		  }[]
+		| undefined
+) {
+	const includePaths: string[][] = [];
+	const excludePaths: string[][] = [];
+
+	let hasPrerendered404 = false;
+	for (const route of routes) {
+		const convertedPath = segmentsToCfSyntax(route.segments, _config);
+		if (route.pathname === '/404' && route.prerender === true) hasPrerendered404 = true;
+
+		if (route.type === 'page') if (route.prerender === false) includePaths.push(convertedPath);
+
+		if (route.type === 'endpoint')
+			if (route.prerender === false) includePaths.push(convertedPath);
+			else excludePaths.push(convertedPath);
+
+		if (route.type === 'redirect') excludePaths.push(convertedPath);
+	}
+
+	for (const page of pages) {
+		const pageSegments = removeLeadingForwardSlash(page.pathname)
+			.split(posix.sep)
+			.filter(Boolean)
+			.map((s) => {
+				return getParts(s);
+			});
+		excludePaths.push(segmentsToCfSyntax(pageSegments, _config));
+	}
+
+	if (existsSync(fileURLToPath(_config.publicDir))) {
+		const staticFiles = await glob(`${fileURLToPath(_config.publicDir)}/**/*`, {
+			cwd: fileURLToPath(_config.publicDir),
+			filesOnly: true,
+			dot: true,
+		});
+		for (const staticFile of staticFiles) {
+			if (['_headers', '_redirects', '_routes.json'].includes(staticFile)) continue;
+			const staticPath = staticFile;
+
+			const segments = removeLeadingForwardSlash(staticPath)
+				.split(posix.sep)
+				.filter(Boolean)
+				.map((s: string) => {
+					return getParts(s);
+				});
+			excludePaths.push(segmentsToCfSyntax(segments, _config));
+		}
+	}
+
+	const assetsPath = segmentsToCfSyntax(
+		[
+			[{ content: _config.build.assets, dynamic: false, spread: false }],
+			[{ content: '', dynamic: true, spread: false }],
+		],
+		_config
+	);
+	excludePaths.push(assetsPath);
+
+	const pagefindPath = segmentsToCfSyntax(
+		[
+			[{ content: 'pagefind', dynamic: false, spread: false }],
+			[{ content: '', dynamic: true, spread: false }],
+		],
+		_config
+	);
+	excludePaths.push(pagefindPath);
+
+	for (const redirect of redirects) {
+		excludePaths.push(segmentsToCfSyntax(redirect, _config));
+	}
+
+	const includeTrie = new PathTrie();
+	for (const includePath of includePaths) {
+		includeTrie.insert(includePath);
+	}
+	const deduplicatedIncludePaths = includeTrie.getAllPaths();
+
+	const excludeTrie = new PathTrie();
+	for (const excludePath of excludePaths) {
+		excludeTrie.insert(excludePath);
+	}
+	const deduplicatedExcludePaths = excludeTrie.getAllPaths();
+
+	if (
+		!hasPrerendered404 ||
+		deduplicatedIncludePaths.length > 100 ||
```

Comment:
Two things:
- lone numbers aren't self-explanatory, it's usually better to assign them to constants so the name helps to give them some context
- comment with a link to the reference explaining the limit. That's a very important piece of information.